packages/@react-spectrum/tag/src/TagGroup.tsx

```diff
@@ -162,6 +170,15 @@ function TagGroup<T extends object>(props: SpectrumTagGroupProps<T>, ref: DOMRef
   let showActions = tagState.showCollapseButton || (actionLabel && onAction);
   let isEmpty = state.collection.size === 0;
 
+  let containerStyle = useMemo(() => {
+    if (maxRows == null || !isCollapsed || !tagsRef.current) {
+      return undefined;
+    }
+    let maxHeight = (TAG_STYLES[scale].height + (TAG_STYLES[scale].margin * 2)) * maxRows;
+    return {maxHeight, overflow: 'hidden'};
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCollapsed, maxRows, tagsRef.current]);
```

Comment:
Bulk of this PR's behavior change is here, and the rest is mostly cleanup.

Reply:
we should remove the tagsRef.current from the dependency array, only state and derived values should be in a dependency array

why do we need to check it, could we check something else instead?

Reply:
I think that was just leftover from measuring there, I can remove that. Thanks